src/sas/qtgui/Perspectives/Fitting/FittingUtilities.py

```diff
@@ -748,14 +748,9 @@ def calcWeightIncrease(weights, ratios, flag=False):
 
     # Calc statistical weight for each dataset and maximum
     # Need to find the best "weighting" scheme
-    min_weight = numpy.inf
-    for id_index in weights.keys():
-        stat_weights[id_index] = numpy.sum(1.0/weights[id_index]**2)
-        if stat_weights[id_index] < min_weight:
-            min_weight = stat_weights[id_index]
-
-    for id_index in weights.keys():
-            weight_increase[id_index] = float(ratios[id_index]) * min_weight / stat_weights[id_index]
+    stat_weights = {k: numpy.sum(v**-2) for k, v in weights.items()}
+    min_weight = min(v for v in stat_weights.values())
```

Comment:
I could have suggested:
```
min_weight = min(stat_weights.values())
```
This is twice as fast! 0.7 μs vs 1.65 μs for five entries 😉

Pick what seems more readable to you.

Reply:
Ah, yes! This is faster and simpler, so I will use your suggestion.

I still have to check the nllf method, and try to generate or find other data sets to test how stable the method is.